js/piwik.js

```diff
@@ -5759,9 +5762,30 @@ if (typeof window.Matomo !== 'object') {
              * @param bool
              */
             this.setSecureCookie = function (enable) {
+                if(location.protocol !== 'https:') {
+                    logConsoleError("Error in setSecureCookie: You cannot use `Secure` on http.");
+                }
                 configCookieIsSecure = enable;
             };
 
+            /**
+             * Set the SameSite attribute for cookies to a custom value.
+             * You might want to use this if your site is running in an iframe since
+             * then it will only be able to access the cookies if SameSite is set to 'None'.
+             * Sets to Lax if invalid parameter is passed, sets CookieIsSecure to true on None.
+             *
+             * @param string either Lax, None or Strict
+             */
+            this.setCookieSameSite = function (sameSite) {
+                if (sameSite != 'None' && sameSite != 'Lax' && sameSite != 'Strict') {
```

Comment:
btw not too important but be great to use `!==` and below `===` for exact comparison.

Also should we maybe compare lowercase and then convert the first letter to upper case when setting the value? Just mentioning this because some users will try to set `lax` or `strict`.

Meaning be like 

```js
this.setCookieSameSite = function (sameSite) {
sameSite = String(sameSite).toLowerCase();
 if (sameSite === 'none' || sameSite === 'lax' || sameSite === 'strict') {
    sameSite = sameSite.charAt(0).toUpperCase() + sameSite.slice(1)
} else {
   logConsoleError('ignored value for sameSite. Please use either lax, none, or strict')
}
```

see above example be also good to not silently convert value to `Lax`

Reply:
If a user has a site available through HTTP and HTTPS, then setting `sameSite('None')` would only work on HTTPS. On HTTP it would log an error but still try to set `sameSite=None`. I wonder if we would need to additionally add something like below

```js
if (sameSite === 'none' && location.protocol !== 'https:') {
   logConsoleError('sameSite=none wont work because it is not using HTTPS')
   return;
}
```

this way we would keep `Lax` and a user can conveniently set `sameSite('None')` and know it will only be applied on HTTPS traffic.
We would still keep the additional ` if(location.protocol !== 'https:') {` check in `setSecureCookie` as this is useful too.

Reply:
As Lax is the Default value, I was reverting to that, so that a non-valid combination can't happen.
Will change in next commit.

Reply:
I don't think it is necessary to revert to Lax on HTTP, as the usecase for Samesite=None will only work on https and the cookie will not be accepted by the browser in that case anyways.